app/post.go

```diff
@@ -393,6 +393,14 @@ func (a *App) FillInPostProps(post *model.Post, channel *model.Channel) *model.A
 		post.DelProp("channel_mentions")
 	}
 
+	if !a.HasPermissionToChannel(post.UserId, post.ChannelId, model.PERMISSION_USE_CHANNEL_MENTIONS) {
+		post.AddProp(model.POST_PROPS_MENTION_HIGHLIGHT_DISABLED, true)
+	}
+
+	if !a.HasPermissionToChannel(post.UserId, post.ChannelId, model.PERMISSION_USE_GROUP_MENTIONS) {
+		post.AddProp(model.POST_PROPS_GROUP_HIGHLIGHT_DISABLED, true)
+	}
```

Comment:
can we make this check a bit smarter? maybe we can do a regex check for `\B@` and only if that exists we check the permission and set the prop?

Reply:
Done! + added tests for this add property portion of the code.